### PR TITLE
Move login logic to api project

### LINKED ISSUE #1461: Factor server-side auth code into its own Laravel subproject
# Description
Create a new Laravel project and move the login functionality from /admin to this new project. That means the `/login` and `/auth-callback` routes, and the AuthController.

This will require some modification of the root htaccess file. Ensure it is still possible to log into the admin dashboard.

### PR DESCRIPTION
**Pull request by petertgiles**
This branch:
- Upgrades the API project from Lumen to Laravel
- Moves the login/logout/refresh stuff from the ADMIN project to the API project
- It's a lot of files to change.  I recommend reviewing commit-by-commit instead.

:warning: This PR changes the way the environment variables are laid out between projects :warning: 
These changes could break your development environment until you update your local .env files.  It is recommend to do a compare/diff between the .env.example and .env file in each project to ensure your settings are up to date.

- API_APP_URL in _api_ has the port number added to the default value
- OAUTH_REDIRECT_URI moved from _admin_ to _api_
- OAUTH_POST_LOGIN_REDIRECT in _api_ is a new variable
- OAUTH_URI moved from _admin_ to _api_
- OAUTH_TOKEN_URI moved from _admin_ to _api_
- OAUTH_ADMIN_CLIENT_ID was renamed to *API* and moved from _admin_ to _api_
- OAUTH_ADMIN_CLIENT_SECRET was named to *API* and moved from _admin_ to _api_

Closes #1461

### CODE REVIEW

File: frontend/admin/resources/js/apiRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const apiRoot = (): string => "/";
+  return {
+    login: (from?: string, locale?: string): string => {
+      const searchTerms: string[] = [];
+      if (from) searchTerms.push(`from=${encodeURI(from)}`);
+      if (locale) searchTerms.push(`locale=${encodeURI(locale)}`);
+      const searchString = searchTerms.reduce((previousValue, currentValue) => {
+        if (previousValue) return `${previousValue}&${currentValue}`;
+        return currentValue;
+      }, "");
```
Comment (tristan-orourke) [inline frontend/admin/resources/js/apiRoutes.ts:15]: I think you might have overthought this 😉  - isn't this just [array.join()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join) ?
```suggestion
      const searchString = searchTerms.join("&");
```

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
This looks great! I was a bit unsure about the "trimming to be more like lumen" commit https://github.com/GCTC-NTGC/gc-digital-talent/pull/2220/commits/782058d06c1333c39cfc128c1027556961aaf458, but we can always add those middleware back in if we need them.

I also spent a bit of time looking at how Laravel recommends [setting up custom auth](https://laravel.com/docs/9.x/authentication#adding-custom-guards), which has a slightly different convention, but looks like our AuthServiceProvider is doing exactly the same thing. Might be something we want to refactor later, maybe take another look for a Laravel OAuth library.

Okay, my read-through code review is done, I'll do my QA tests tomorrow morning. 👍

**Comment by tristan-orourke** [comment]
Oops, I meant to just leave a comment, not request changes. No changes requested yet! 😆

File: frontend/admin/resources/js/apiRoutes.ts (hunk 1 above)
Comment (petertgiles) [reply frontend/admin/resources/js/apiRoutes.ts]: Haha, you're right!  I was originally trying to reduce a collection of key-value pairs but gave up.  I didn't take the next logical simplification step.

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Works as advertised! Except the logout uri vars need to stay in the frontend/admin env.

I appreciate @patcon's suggestion about the OAUTH variables message, and how you've implemented it. Since it references this pull request, it might be helpful having a few more notes in the PR description about the specifics of migrating env variables. One "gotcha" that got me was forgetting to rename OAUTH_ADMIN_CLIENT_ID/SECRET to OAUTH_API_CLIENT_ID/SECRET, as I initially tried copying them over directly.

File: api/.env.example [hunk 2]
```diff
@@ -15,12 +17,58 @@ DB_DATABASE=gctalent
...
+PUSHER_APP_ID=
+PUSHER_APP_KEY=
+PUSHER_APP_SECRET=
+PUSHER_APP_CLUSTER=mt1
+
+MIX_PUSHER_APP_KEY="${PUSHER_APP_KEY}"
+MIX_PUSHER_APP_CLUSTER="${PUSHER_APP_CLUSTER}"
+
+OAUTH_REDIRECT_URI="http://localhost:8000/auth-callback"
+OAUTH_POST_LOGOUT_REDIRECT="http://localhost:8000/admin"
```
[the remark this answers is not in the document]
Comment (petertgiles) [reply api/.env.example]: Great catch!  I've fixed it.

**Comment by petertgiles** [comment]
> it might be helpful having a few more notes in the PR description

Great call out.  I've added them.

**Review by tristan-orourke — APPROVED** [review]
Perfect, thanks for the detailed env variable instructions!

**Pull request merged**